apps/examples/mediarecorder/mediarecorder_main.cpp

```diff
@@ -234,29 +245,36 @@ class MediaRecorderTest : public MediaRecorderObserverInterface
 			return;
 		}
 
-		mp.setObserver(shared_from_this());
-		mp.setDataSource(std::move(source));
-
-		if (mp.prepare() == PLAYER_OK) {
-			isPlaying = true;
-			if (mp.start() == PLAYER_OK) {
-				// playback started
-				return;
-			}
+		if (mp.setObserver(shared_from_this()) == PLAYER_ERROR) {
+			std::cout << "Mediaplayer::setObserver failed" << std::endl;
+			goto errout_destroy;
+		}
 
-			std::cout << "Mediaplayer::start failed" << std::endl;
-			isPlaying = false;
+		if (mp.setDataSource(std::move(source)) == PLAYER_ERROR) {
+			std::cout << "Mediaplayer::setDataSource failed" << std::endl;
+			goto errout_destroy;
+		}	
 
-			if (mp.unprepare() == PLAYER_ERROR) {
-				std::cout << "Mediaplayer::unprepare failed" << std::endl;
-			}
-		} else {
+		if (mp.prepare() == PLAYER_ERROR) {
 			std::cout << "Mediaplayer::prepare failed" << std::endl;
+			goto errout_destroy;
 		}
 
+		if (mp.start() == PLAYER_ERROR) {
+			std::cout << "Mediaplayer::start failed" << std::endl;
+			goto errout_unprepare;
+		}
+		return;
+
+	errout_unprepare:
+		if (mp.unprepare() == PLAYER_ERROR) {
+			std::cout << "Mediaplayer::unprepare failed" << std::endl;
+		}
+	errout_destroy:
 		if (mp.destroy() == PLAYER_ERROR) {
 			std::cout << "Mediaplayer::destroy failed" << std::endl;
 		}
+		return;
```

Comment:
Above return & here will make execute application only once, we need to test multiple time.

Reply:
This return is the end of the play_data function.
It does not seem to be a problem.